### PR TITLE
Update `postcss` to v8

### PR DESCRIPTION
**Pull request by fisker**
- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-css/parser-postcss.js [hunk 1]
```diff
@@ -300,6 +300,33 @@ function parseNestedCSS(node, options) {
...
+            node.source.end.offset + 1
+          );
+        const fakeContent = textBefore.replace(/[^\n]/g, " ") + nodeText;
+        node.value = {
+          type: "css-rule",
+          nodes: parseCss(fakeContent, [], { ...options }).nodes[0].nodes,
+        };
+        return node;
+      }
+    }
```
Comment (evilebottnawi) [inline src/language-css/parser-postcss.js:328]: I think we should remove them in the future major release, it is invalid

File: tests/css/comments/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -887,11 +925,11 @@ article /* comment 168 */ :--heading /* comment 169 */ + /* comment 170 */ p /*
 /* custom properties set & @apply rule */
 :root {
   /* comments 192 */
-  --centered /* comments 193 */ : /* comments 194 */ {
+  --centered/* comments 193 */ : /* comments 194 */ {
     display: flex;
     align-items: center;
     justify-content: center;
-  }
+  };
```
Comment (evilebottnawi) [inline tests/css/comments/__snapshots__/jsfmt.spec.js.snap:932]: Regression?

File: src/language-css/parser-postcss.js (hunk 1 above)
Comment (fisker) [reply src/language-css/parser-postcss.js:328]: I was going to parse is as `value-unknown`, but seems it is a proposal, but abandoned

http://tabatkins.github.io/specs/css-apply-rule/#defining

File: tests/css/comments/__snapshots__/jsfmt.spec.js.snap (hunk 2 above)
Comment (fisker) [reply tests/css/comments/__snapshots__/jsfmt.spec.js.snap:932]: Acutally, I think it's bugfix.

`--centered` is a prop, it should add `;`, the space before `/* comments 193 */` is gone, it's because we print `decl` like this, [playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBDABMAOld7IA2EATugPQBU+EAtjQjOhWUugMQBmADFwNzo4AviAA0ICAAcYAS2gBnZKFTFiEAO4AFZQgUpUBNagCeCsQCNiqMAGs4MAMoSr0qAHNkMYgFc4YuDTM4ABMg4IAZVDcvVFc4ADESGlQYGTdkNC8YCFEQAAsYGgIAdVzpeDknMDh7HTLpADcyo3SwOVMQFzk4YhgNS1ck5A59LrEAKzkADwAhSxs7e1R6MJc4IZHfEAnJ+xdXAjgARS8IeHWCUZAnYi7iFraciWIXGCLpIJhc5AAOLjEniBdIqWCTpJ5wW71NZiACOJ3gfUkujQcgAtFA4MFgjliHA4dJcX0YoMkMMLpsujRpB5vBS9gdjqc1qSNmIYKgzG8Pl8kAAmNmWaQEPYAYVoJJAEIArDkvF0ACoc3Rky71HwASSgoVg9jAzykAEEtfYYEYDucuoJBEA).

There is a space, because we treat as `rule`'s `selector` before, [playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBDABAegFTsgW3wRnW03WAB0p08IAbCAJyXQGIAzABi4G51qAXxAAaEBAAOMAJbQAzslComTCAHcACsoQKUqemtQBPBWIBGTVGADWcGAGUJV6VADmyGEwCucMXHxmcAAmQcEAMqhuXqiucABizPioMDJuyGheMBCiIAAWMPj0AOq50vByTmBw9jpl0gBuZUbpYHKmIC5ycEwwGpauScgc+l1iAFZyAB4AQpY2dvaoRGEucEMjviATk-YurvRwAIpeEPDr9KMgTkxdTC1tORJMLjBF0kEwucgAHFxiTxAukVLBJ0k84Ld6msxABHE7wPqSXRoOQAWigcGCwRyTDgcOkuL6MUGSGGF02XXw0g83gpewOx1Oa1JGzEMFQZjeHy+SAATGzLNJ6HsAMIQQiodIQgCsOS8XQAKhzdGTLvUfABJKChWD2MDPKQAQW19hgRgO5y6gkEQA).

Comment (fisker) [reply tests/css/comments/__snapshots__/jsfmt.spec.js.snap:932]: And the test is added by you, the input has `;` https://github.com/prettier/prettier/blob/e857c4e88ffbf0c2fb213ee7172fbe9431ab87a8/tests/css/comments/selectors.css#L156

Comment (fisker) [reply tests/css/comments/__snapshots__/jsfmt.spec.js.snap:932]: I think we should print `less` and `scss` like this too.

File: src/language-css/parser-postcss.js (hunk 1 above)
Comment (evilebottnawi) [reply src/language-css/parser-postcss.js:328]: Yes, but it is violate CSS spec, so prefer to avoid this, because it can break some future CSS features

File: tests/css/comments/__snapshots__/jsfmt.spec.js.snap (hunk 2 above)
Comment (evilebottnawi) [reply tests/css/comments/__snapshots__/jsfmt.spec.js.snap:932]: scss (need test on less) supports nested declarations https://www.sassmeister.com/, so prefer keep it as is

Comment (fisker) [reply tests/css/comments/__snapshots__/jsfmt.spec.js.snap:932]: https://www.sassmeister.com/ add ';' too.

Comment (fisker) [reply tests/css/comments/__snapshots__/jsfmt.spec.js.snap:932]: http://lesscss.org/less-preview/

Can parse

```less
:root {
    /* comments 192 */ --centered /* comments 193 */ : /* comments 194 */ {
        display: flex;
        align-items: center;
        justify-content: center;
    };
}
```

But can't parse without `;`

```less
:root {
    /* comments 192 */ --centered /* comments 193 */ : /* comments 194 */ {
        display: flex;
        align-items: center;
        justify-content: center;
    }
}
```

Comment (evilebottnawi) [reply tests/css/comments/__snapshots__/jsfmt.spec.js.snap:932]: Remove `;` and no `;` in output

Comment (evilebottnawi) [reply tests/css/comments/__snapshots__/jsfmt.spec.js.snap:932]: For less we should not change broken output, it is AST modification, so for less and sass better to keep input as is

File: tests/css/postcss-8-improment/__snapshots__/jsfmt.spec.js.snap [hunk 3]
```diff
@@ -32,7 +78,7 @@ https://github.com/postcss/postcss/releases/tag/8.0.0
 */
 
 :root {
-  --empty:  ;
+  --empty: ;
```
Comment (evilebottnawi) [inline tests/css/postcss-8-improment/__snapshots__/jsfmt.spec.js.snap:81]: Good catch :+1:

Comment (evilebottnawi) [reply tests/css/postcss-8-improment/__snapshots__/jsfmt.spec.js.snap:81]: `--empty:;` is invalid syntax
`--empty: ;` is valid syntax

So both should not be changed

File: tests/css/comments/__snapshots__/jsfmt.spec.js.snap (hunk 2 above)
Comment (evilebottnawi) [reply tests/css/comments/__snapshots__/jsfmt.spec.js.snap:932]: Can we add test without `;`?

File: tests/css/postcss-8-improment/__snapshots__/jsfmt.spec.js.snap [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  --JSON: [1, "2", {"three": {"a": 1}}, [4]];
+  --javascript: function(rule) {console.log(rule)};
+}
+
+@supports (--element(".minwidth", {"minWidth": 300})) {
+  [--self] {
+    background: greenyellow;
+  }
+}
```
Comment (evilebottnawi) [inline tests/css/postcss-8-improment/__snapshots__/jsfmt.spec.js.snap:68]: Can we add `fixes` to the PR message and add changelog? Or you can do it in the separate PR?

**Comment by evilebottnawi** [comment]
@fisker Ready for review?

**Comment by fisker** [comment]
Yes, please review.

File: tests/css/comments/__snapshots__/jsfmt.spec.js.snap (hunk 2 above)
Comment (evilebottnawi) [reply tests/css/comments/__snapshots__/jsfmt.spec.js.snap:932]: and we should not fix invalid syntax, it is ast modification

Comment (fisker) [reply tests/css/comments/__snapshots__/jsfmt.spec.js.snap:932]: @evilebottnawi I only changed the output for css,  it's custom property, it should add `;` if it's last one, like other properties. The output for less and scss didn't change. I've add tests.

The following one is not adding `;` 

**Prettier pr-9209**
[Playground link](https://deploy-preview-9209--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuZAnCEYAJgB0pZYC0RAZhkngYVpADYSpJYDEpADOwNz6EC+vYkQCGUTAAs4TVp1I8ofEABoQEAA4wAltADOyUMNToA7gAVDCPSmF1jwgJ56VAI1TCwAazgwAymveaUADmyDCoAK5wKnAAts5wACYJiQAyokHhwkFwAGKMMcIwWsHIIMLhMBDKIOIwMXQA6uKa8Dr+YHA+li2aAG4t9qVgOk4ggTpSMKZuQQXIpDYTKgBWOgAeAEJunt4+wjFwKYFw84tRIKtrPoFBdHAAiuGYJ0gLdEsg-qgTqEMj1WpUIEYA1NAkYOJkAAOdgqQEQCYNNxqUqAuA-XonFQARye8Gm6isZR0RCgcESiWqqDguM01OmWTmrzOKgmMU0oQi5x0Nzuj2ep3e5xgwmcoPBkKQACYVGFhJo6DcAMIQGJMkDogCs1XCEwAKqKrG8Pr1IgBJKDJWA+MBAjQAQUtPhg9jugomfD4QA)
```sh
--parser css
```

**Input:**
```css
:root {
  --foo:{
    color: #f00;
  }
  --another: #00f;
}
```

**Output:**
```css
:root {
  --foo: {color: #f00;} --another: #00f;
}

``` 
WDYT?

Comment (evilebottnawi) [reply tests/css/comments/__snapshots__/jsfmt.spec.js.snap:932]: I think we should not put it in one file, it is like declarations, so better to have each declaration on own line

Comment (fisker) [reply tests/css/comments/__snapshots__/jsfmt.spec.js.snap:932]: The whole part after `--foo` is parsed as value. And sent to `value-parser`, how do we tell this should not parse as `value`?

Comment (fisker) [reply tests/css/comments/__snapshots__/jsfmt.spec.js.snap:932]: Starts with `{`?

Comment (evilebottnawi) [reply tests/css/comments/__snapshots__/jsfmt.spec.js.snap:932]: It we have `--` at the start and `: {`

Comment (fisker) [reply tests/css/comments/__snapshots__/jsfmt.spec.js.snap:932]: Okay, I'll change this  case to  let the value parse  as  `value-unknown`

Comment (fisker) [reply tests/css/comments/__snapshots__/jsfmt.spec.js.snap:932]: Fixed, https://github.com/prettier/prettier/pull/9209/commits/c5e3bdd0e1f41dba79d1fab3866283c634b0d8e6.

**Review by evilebottnawi — APPROVED** [review]
Looks good, in the next major release we should print `--var: { }` as is do not touch this, because it is violate CSS spec

**Review by sosukesuzuki — APPROVED** [review]
